pkg/discovery/kv/kv_test.go

```diff
@@ -122,6 +111,65 @@ func (ds *DiscoverySuite) TestWatch(c *check.C) {
 	close(stopCh)
 	c.Assert(<-ch, check.IsNil)
 	c.Assert(<-errCh, check.IsNil)
+}
+
+type FakeStore struct {
+	Endpoints  []string
+	Options    *store.Config
+	mockKVChan <-chan []*store.KVPair
+
+	watchTreeCallCount int
+}
+
+func (s *FakeStore) Put(key string, value []byte, options *store.WriteOptions) error {
+	return nil
+}
+
+func (s *FakeStore) Get(key string) (*store.KVPair, error) {
+	return nil, nil
+}
+
+func (s *FakeStore) Delete(key string) error {
+	return nil
+}
+
+func (s *FakeStore) Exists(key string) (bool, error) {
+	return true, nil
+}
+
+func (s *FakeStore) Watch(key string, stopCh <-chan struct{}) (<-chan *store.KVPair, error) {
+	return nil, nil
+}
+
+func (s *FakeStore) WatchTree(directory string, stopCh <-chan struct{}) (<-chan []*store.KVPair, error) {
+	fmt.Printf("%s\n", directory)
```

Comment:
Oh I should remove that line though... And add comments on the other to say that it's for implementing `Store` and as it's not tested, it does nothing :cat:.